src/python/grpcio/grpc/_server.py

```diff
@@ -880,7 +880,11 @@ def _serve(state):
         if state.server_deallocated:
             _begin_shutdown_once(state)
         if event.completion_type != cygrpc.CompletionType.queue_timeout:
-            if not _process_event_and_continue(state, event):
+            try:
+                if not _process_event_and_continue(state, event):
+                    return
+            except RuntimeError:
```

Comment:
Should we try to catch more exception types?

Reply:
yes, `except Exception` instead of `except RuntimeError`